### PR TITLE
Add tax year 2023 data

### LINKED ISSUE #39: Update tax and PTAXSIM with 2023 data
Complete 2023 tax data is now available from the Clerk and Treasurer databases! Please update `tax` and then PTAXSIM.

### PR DESCRIPTION
**Pull request by dfsnow**
Closes #39.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -2,6 +2,6 @@ message: "If you use this software, please cite it as below."
 authors:
 - family-names: "Cook County Assessor's Office"
 title: "PTAXSIM"
-version: 0.6.1
+version: 0.6.2
```
Comment (dfsnow) [inline CITATION.cff:5]: Bumping the package version solely due to the addition of 2023 testing data. No actual package functionality has changed.

File: DESCRIPTION [hunk 2]
```diff
@@ -51,7 +51,7 @@ Suggests:
     sf,
     snakecase,
     stringr,
-    tabulizer,
+    tabulapdf,
```
Comment (dfsnow) [inline DESCRIPTION:54]: Very annoyingly, `tabulizer` is fully deprecated and now replaced by `tabulapdf`. The changes in `sample_tax_bills_detail.R` are mostly related to accounting for the slight behavior difference between the two packages.

File: _pkgdown.yml
Comment (dfsnow) [inline _pkgdown.yml:1]: No content here has changed (only whitespace) except for bumping the bootstrap theme to v5.

**Review by jeancochrane — APPROVED** [review]
Looks great! I'm not really sure how to test these changes but I trust that you've done so.

**Pull request merged**